redmine_import_time_entry/models/redmine_backend.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# © 2016 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.tools.translate import _
+from odoo.tools import ustr
+from odoo.exceptions import UserError
+
+from datetime import datetime, timedelta
+
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+to_string = fields.Date.to_string
+
+
+class redmine_backend(models.Model):
+    _inherit = 'redmine.backend'
+
+    contract_ref = fields.Char(
+        'Contract # field name',
+        help="The field in Redmine used to relate a project in Redmine "
+        "to a project in Odoo. Each redmine project must have a unique "
+        "value for this attribute."
+    )
+    sync_tasks = fields.Boolean(
+        "Sync Versions -> Tasks",
+        help="Redmine Versions will be synchronized with Odoo tasks, "
+             "using their name to match")
+    time_entry_last_update = fields.Datetime(
+        'Last Time Entry Update', required=True,
+        # At the first import, this field must have a value so that the
+        # update_on field on Redmine time entries can be compared to it
+        # Comparing False with a datetime object raises an error.
+        default=lambda self: datetime(1900, 1, 1)
+    )
+    time_entry_number_of_days = fields.Integer(
+        'Time Entries - Number of days',
+        help="Number of days used when fetching the time entries.",
+        required=True,
+        default=14,
+    )
+
+    @api.multi
+    def check_contract_ref(self):
+        """
+        Check if the contract_ref field exists in redmine
+        """
+        self.ensure_one()
+        with self.work_on('redmine.backend') as work:
+            adapter = work.component(usage='backend.adapter')
+
+        try:
+            adapter._auth()
+        except Exception as e:
+            raise UserError(
+                'Could not connect to Redmine: %s' % ustr(e))
+
+        projects = adapter.redmine_api.project.all()
```

Comment:
This way checks also if the value is set for the custom field in redmine.

```diff
         projects = adapter.redmine_api.project.all()
         exist = False
+        assigned = False
 
         if projects:
             if hasattr(projects[0], 'custom_fields'):
                 for cs in projects[0].custom_fields:
                     if cs['name'] == self.contract_ref:
                         exist = True
+                        assigned = bool(cs['value'])
             elif hasattr(projects[0], self.contract_ref):
                 exist = True
 
-        if exist is True:
-            raise UserError(
-                _('Connection test succeeded\n'
-                  'Everything seems properly set up'))
-        else:
+        if not exist:
             raise UserError(
                 _("Redmine backend configuration error\n"
                   "The contract # field name doesn't exist.")
             )
+        elif not assigned:
+            raise UserError(
+                _("Redmine backend configuration error\n"
+                  "The contract # field is not assigned in Redmine.")
+            )
+        else:
+            raise UserError(
+                _('Connection test succeeded\n'
+                  'Everything seems properly set up'))
```
Sorry if I did mistake the format. I am learning to make reviews. :-)